onnx/defs/tensor/defs.cc

```diff
@@ -2895,8 +2896,12 @@ ONNX_OPERATOR_SET_SCHEMA(
           // and 1 element vector for now. In future when version update for
           // this op is done we should only allow scalar or change the spec to
           // allow both.
+          std::optional<int64_t> depth_value;
           if (hasInputShape(ctx, 1)) {
             auto& depth_shape = getInputShape(ctx, 1);
+            if (const TensorProto *depth_data = ctx.getInputData(1)) {
+              depth_value = ParseData<int64_t>(depth_data)[0];
+            }
```

Comment:
It might not be int64. Any suggestion to easily handle int8/16/32/64 data?

Reply:
Check the type and handle it case-by-case: eg., like [here](https://github.com/onnx/onnx/blob/d7e2b813c1a26cdb8bcb6c4547cb5e6ffeb9b49b/onnx/defs/sequence/defs.cc#L347) ? I guess it would be nice to create an utility function others can also use.

Reply:
I realized that [onnxruntime is now supporting only float/int32/int64 for depth](https://github.com/microsoft/onnxruntime/blob/cd56ea4a74ee41c040899d702667d2c86bee4ef0/onnxruntime/core/providers/cpu/tensor/onehot.cc#L59-L69).